frontend/src/metabase/common/components/Table/ControlledTable.tsx

```diff
@@ -0,0 +1,149 @@
+import React from "react";
+
+import { PaginationControls } from "metabase/components/PaginationControls";
+import { color } from "metabase/lib/colors";
+import { Box, Flex, Icon } from "metabase/ui";
+
+import CS from "./Table.module.css";
+
+export type BaseRow = Record<string, unknown> & { id: number | string };
+
+type ColumnItem = {
+  name: string;
+  key: string;
+  sortable?: boolean;
+};
+
+type sortProps = { name: string; direction: "asc" | "desc" };
+
+export type ControlledTableProps<Row extends BaseRow> = {
+  columns: ColumnItem[];
+  rows: Row[];
+  rowRenderer: (row: Row) => React.ReactNode;
+  tableProps?: React.HTMLAttributes<HTMLTableElement> & { className: string };
+  sortColumn?: sortProps;
+  onSort?: (props: sortProps) => void;
+  onPageChange?: (page: number) => void;
+  page?: number;
+  pageSize?: number;
+  totalItems?: number;
+  ifEmpty?: React.ReactNode;
+  cols?: React.ReactNode;
+};
+
+/**
+ * A basic reusable table component that supports client-side sorting by a column
+ *
+ * @param columns     - an array of objects with name and key properties
+ * @param rows        - an array of objects with keys that match the column keys
+ * @param rowRenderer - a function that takes a row object and returns a <tr> element
+ * @param tableProps  - additional props to pass to the <table> element
+ */
+
+export function ControlledTable<Row extends BaseRow>({
+  columns,
+  rows,
+  rowRenderer,
+  sortColumn,
+  onSort,
+  page,
+  pageSize,
+  onPageChange,
+  totalItems,
+  ifEmpty,
+  cols,
+  ...rest
+}: ControlledTableProps<Row>) {
+  return (
+    <>
+      <table {...rest} className={CS.Table}>
+        {cols && <colgroup>{cols}</colgroup>}
+        <thead>
+          <tr>
+            {columns.map(column => (
+              <th key={String(column.key)}>
+                {onSort && column.sortable !== false ? (
+                  <ColumnHeader
+                    column={column}
+                    sortColumn={sortColumn?.name}
+                    sortDirection={sortColumn?.direction}
+                    onSort={(columnKey: string, direction: "asc" | "desc") => {
```

Comment:
I think we have a type for `"asc" | "desc"` that Raffi made somewhere

Reply:
Yeah, I changed to the enum types